### DISCUSSION TITLE
Deployment of Keycloak over a multi-staging enviornment

### DISCUSSION THREAD

**Discussion by ThomasAppDev** [keycloak/keycloak#22720] [Q&A]
Hello,

we are planning to deploy keycloak in a multi-staging environment:
- test: try out a new configuration
- stage: as near as possible as production including the integration of clients
- production: Real Environment

Our deployment itself is done with ansible and the keycloak docker container. We wanted to do the configuration on the test-system, export the realm, then test the deployment by importing it again in the stage and if everything works deploy it to production.

As we tried it out, the override of the realm-configration in staging, erased all client-configurations. How do you do a multi-Stage deployment? Is there an option to only override the functions, that are in the realm-export included?

Thank you for you help!

**Comment by DanielKucal** [comment]
Up, I'm looking for similar answers in 2024

**Comment by rene6502** [comment]
I can recommend to use https://github.com/adorsys/keycloak-config-cli for such scenario.
